### PR TITLE
added support for aws rds group option 

### PR DESCRIPTION
**Pull request by Nirbhay1997**
Signed-off-by: nirbhay <nkumar@chef.io>

### Description

Please describe what this change achieves. Ensure you have read the [Contributing to InSpec AWS](https://github.com/inspec/inspec-aws/CONTRIBUTING.md) document before submitting.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/aws_rds_group_options.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+title: About aws_rds_group_options Resource
+---
+
+# aws\_rds\_group\_options
+
+Use the `aws_rds_group_options` InSpec audit resource to test detailed properties of an individual aws_rds_group_options.
```
Comment (IanMadd) [inline docs/resources/aws_rds_group_options.md:7]:
```suggestion
Use the `aws_rds_group_options` InSpec audit resource to test detailed properties of a group of Amazon Relational Database Service (RDS) clusters.
```

File: docs/resources/aws_rds_group_options.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+it { should exist }
+end
+
+    # Can also use hash syntax
+    describe aws_rds_group_options do
+      it { should exist }
+    end
+
+#### Parameters
+
```
Comment (IanMadd) [inline docs/resources/aws_rds_group_options.md:24]: We should really have a table of available parameters in here.

File: docs/resources/aws_rds_group_option.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+##### option\_group\_name or engine\_name _(required)_
+
+This resource accepts a single parameter, the user-supplied cluster identifier. This parameter isn't case-sensitive.
+This can be passed either as a string or as a `option_group_name: 'value'` key-value entry in a hash.
+
+See also the [AWS documentation on RDS cluster](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Overview.html).
+
+## Properties
+
```
Comment (IanMadd) [inline docs/resources/aws_rds_group_option.md:33]: We should have a table of available properties in here.

File: libraries/aws_rds_option_group.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    param = { option_group_name: opts[:option_group_name] }
+
+    catch_aws_errors do
+      @display_name = opts[:option_group_name]
+
+      begin
+        resp = @aws.rds_client.describe_option_groups(param)
+        return if resp.option_groups_list.empty?
+        @rds_cluster = resp.option_groups_list[0].to_h
+      rescue Aws::RDS::Errors::OptionGroupNotFoundFault
```
Comment (sathish-progress) [inline libraries/aws_rds_option_group.rb:29]: no need to rescue as it inherits from AWS::ServiceError

File: libraries/aws_rds_option_group.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsRdsOptionGroup < AwsResourceBase
+  name 'aws_rds_option_group'
+  desc 'Verifies settings for an RDS Cluster'
+
+  example "
+    describe aws_rds_option_group(option_group_name: 'test-option_group_name') do
```
Comment (sathish-progress) [inline libraries/aws_rds_option_group.rb:10]: please add one more example without param key!

File: libraries/aws_rds_option_group.rb [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        @rds_cluster = resp.option_groups_list[0].to_h
+      rescue Aws::RDS::Errors::OptionGroupNotFoundFault
+        return
+      end
+      create_resource_methods(@rds_cluster)
+    end
+  end
+
+  def exists?
+    !@rds_cluster.to_s.empty?
```
Comment (sathish-progress) [inline libraries/aws_rds_option_group.rb:37]:
```suggestion
   !!@rds_cluster && !@rds_cluster.empty?
```

File: docs/resources/aws_rds_group_option.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+# aws\_rds\_option\_group
+
+Use the `aws_rds_group_option` InSpec audit resource to test detailed properties of an individual RDS cluster.
+
+RDS gives you access to the capabilities of a MySQL, MariaDB, PostgreSQL, Microsoft SQL Server, Oracle, or Amazon Aurora database server.
+
+## Syntax
+
+An `aws_rds_group_option` resource block uses resource parameters to search for an RDS option group, and then tests that RDS option group.  If no RDS option group match, no error is raised, but the `exists` matcher will return `false` and all properties will be `nil`.
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:13]: Change the line 13  bfefore ##Syntax

File: docs/resources/aws_rds_group_option.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+# aws\_rds\_option\_group
+
+Use the `aws_rds_group_option` InSpec audit resource to test detailed properties of an individual RDS cluster.
+
+RDS gives you access to the capabilities of a MySQL, MariaDB, PostgreSQL, Microsoft SQL Server, Oracle, or Amazon Aurora database server.
+
+## Syntax
+
+An `aws_rds_group_option` resource block uses resource parameters to search for an RDS option group, and then tests that RDS option group.  If no RDS option group match, no error is raised, but the `exists` matcher will return `false` and all properties will be `nil`. 
+    describe aws_rds_group_option(engine_name: 'mysql') do
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:14]: Write the title as Ensure if the engine name exist in the rds group option.

File: docs/resources/aws_rds_group_option.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    # Can also use hash syntax
+    describe aws_rds_group_option(option_group_name: 'default:aurora-5-6') do
+      it { should exist }
+    end
+
+#### Parameters
+
+##### option\_group\_name or engine\_name _(required)_
+
+This resource accepts a single parameter, the user-supplied cluster identifier. This parameter isn't case-sensitive.
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:27]: Not required.

File: docs/resources/aws_rds_group_option.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    end
+
+    # Can also use hash syntax
+    describe aws_rds_group_option(option_group_name: 'default:aurora-5-6') do
+      it { should exist }
+    end
+
+#### Parameters
+
+##### option\_group\_name or engine\_name _(required)_
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:25]: `option\_group\_name` and `engine\_name` _(required)_

File: docs/resources/aws_rds_group_option.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+This resource accepts a single parameter, the user-supplied cluster identifier. This parameter isn't case-sensitive.
+This can be passed either as a string or as a `option_group_name: 'value'` key-value entry in a hash.
+
+See also the [AWS documentation on RDS cluster](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Overview.html).
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:30]: Change the link

Comment (soumyo13) [reply docs/resources/aws_rds_group_option.md:30]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-optiongroup.html

Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:30]: RDS Option Group

File: docs/resources/aws_rds_group_option.md [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+
+## Properties
+
+For a comprehensive list of properties available to test on an RDS cluster see the [AWS Response Object](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/RDS/DBCluster.html).
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:34]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-optiongroup.html

Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:34]: RDS Option Group

File: docs/resources/aws_rds_group_option.md [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+##### Test the engine used with an aws_rds_group_option
+
+    describe aws_rds_group_option(engine_name: 'mysql') do
+      its('engine_name')         { should eq 'mysql' }
+      its('major_engine_version') { should eq '5.6.37' }
+    end
+
+##### Test the options to allocated to an aws_rds_group_option
+
+    describe aws_rds_group_option(engine_name: 'mysql') do
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:47]: option_group_name is req

File: docs/resources/aws_rds_group_option.md [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+
+See also the [AWS documentation on RDS cluster](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Overview.html).
+
+## Properties
+
+For a comprehensive list of properties available to test on an RDS cluster see the [AWS Response Object](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/RDS/DBCluster.html).
+
+## Examples
+
+##### Test the engine used with an aws_rds_group_option
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:38]: option_group_name is req

File: docs/resources/aws_rds_group_option.md [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+
+This InSpec audit resource has the following special matchers. For a full list of available matchers, please visit our [matchers page](https://www.inspec.io/docs/reference/matchers/).
+
+#### exist
+
+The control will pass if the describe returns at least one result.
+
+Use `should_not` to test the entity should not exist.
+
+    describe aws_rds_group_option(engine_name: 'mysql') do
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:64]: option_group_name is req

File: docs/resources/aws_rds_group_option.md [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+      it { should exist }
+    end
+
+    describe aws_rds_group_option(engine_name: 'mysql') do
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:68]: option_group_name is req

File: docs/resources/aws_rds_group_option.md [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+      it { should_not exist }
+    end
+
+## AWS Permissions
+
+Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `rds:DescribeDBclusters` action with Effect set to Allow.
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:74]: Change the text

File: docs/resources/aws_rds_group_option.md [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+
+You can find detailed documentation at [Actions, Resources, and Condition Keys for Amazon RDS](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonrds.html).
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_option.md:76]: Change the text

File: docs/resources/aws_rds_group_options.md [hunk 19]
```diff
@@ -0,0 +1,91 @@
+---
+title: About aws_rds_group_options Resource
+---
+
+# aws\_rds\_group\_options
```
Comment (soumyo13) [inline docs/resources/aws_rds_group_options.md:5]: Check the comments in the singular resource.

File: libraries/aws_rds_option_group.rb [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  def to_s
+    "RDS option groups #{@display_name}"
```
Comment (soumyo13) [inline libraries/aws_rds_option_group.rb:41]: RDS option groups:

Comment (soumyo13) [reply libraries/aws_rds_option_group.rb:41]: It should be RDS option group

**Comment by sathish-progress** [comment]
please add doc links to readme file

File: docs/resources/aws_rds_group_options.md [hunk 21]
```diff
@@ -0,0 +1,91 @@
...
+end
+
+    # Can also use hash syntax
+    describe aws_rds_group_options do
+      it { should exist }
+    end
+
+#### Parameters
+
+|Property                     | Description|
```
Comment (sathish-progress) [inline docs/resources/aws_rds_group_options.md:25]: please add field names here to the respective property

File: libraries/aws_rds_option_group.rb [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+class AwsRdsOptionGroup < AwsResourceBase
+  name 'aws_rds_option_group'
+  desc 'Verifies settings for an RDS Option Group'
+
+  example "
+    describe aws_rds_option_group(option_group_name: 'test-option_group_name') do
+      it { should exist }
+    end
+
+    describe aws_rds_option_group(option_group_name: 'test-option_group_name') do
```
Comment (sathish-progress) [inline libraries/aws_rds_option_group.rb:14]:
```suggestion
    describe aws_rds_option_group('test-option_group_name') do
```

File: libraries/aws_rds_option_group.rb [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+  def initialize(opts = {})
+    opts = { option_group_name: opts } if opts.is_a?(String)
+    super(opts)
+    validate_parameters(require_any_of: %i(option_group_name))
+
+    raise ArgumentError, "#{@__resource_name__}: .option_group_name must be a string" unless opts.key?(:option_group_name)
+    param = { option_group_name: opts[:option_group_name] }
+
+    catch_aws_errors do
+      @display_name = opts[:option_group_name]
```
Comment (sathish-progress) [inline libraries/aws_rds_option_group.rb:27]: this need not be in this block

File: libraries/aws_rds_option_group.rb [hunk 24]
```diff
@@ -0,0 +1,44 @@
...
+    describe aws_rds_option_group(option_group_name: 'test-option_group_name') do
+      it { should exist }
+    end
+  "
+  def initialize(opts = {})
+    opts = { option_group_name: opts } if opts.is_a?(String)
+    super(opts)
+    validate_parameters(require_any_of: %i(option_group_name))
+
+    raise ArgumentError, "#{@__resource_name__}: .option_group_name must be a string" unless opts.key?(:option_group_name)
```
Comment (sathish-progress) [inline libraries/aws_rds_option_group.rb:23]: is being string the only criteria ?

File: libraries/aws_rds_option_groups.rb [hunk 25]
```diff
@@ -0,0 +1,50 @@
...
+    super(opts)
+    validate_parameters
+    @table = fetch_data
+  end
+
+  def fetch_data
+    option_group_rows = []
+    pagination_options = {}
+    catch_aws_errors do
+      response = @aws.rds_client.describe_option_groups(pagination_options)
```
Comment (sathish-progress) [inline libraries/aws_rds_option_groups.rb:36]: this is a [pageable](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/RDS/Client.html#describe_option_groups-instance_method) response

File: test/integration/verify/controls/aws_rds_option_group.rb [hunk 26]
```diff
@@ -0,0 +1,14 @@
+title 'Test single AWS Option Group'
+
+aws_option_group_name= attribute(:aws_db_option_group_name, default: '', description: 'The AWS option group name.')
+
+describe aws_rds_option_group(option_group_name: aws_option_group_name ) do
+  it { should exist }
+  its('option_group_name')                           { should eq 'option-group-test-terraform2' }
+  its('engine_name')                                 { should eq 'sqlserver-ee' }
```
Comment (sathish-progress) [inline test/integration/verify/controls/aws_rds_option_group.rb:8]: why not retrieve engine names via attributes?

**Review by sathish-progress — APPROVED** [review]
LGTM

**Pull request merged**